### PR TITLE
sock_diag: wire up

### PR DESCRIPTION
**Pull request by george-hopkins**
As I needed to quickly diagnose some sockets, I took the liberty to finish up your work in #56. Specifically, I added a buffer for the messages and an example. Other than a small fix of the serialization of port numbers, it's just a rebased version of your implementation. I'm happy to close this PR and rebase my fixes on your work in case you'd like to keep working on it.

### CODE REVIEW

**Comment by little-dude** [comment]
Hi @george-hopkins this is very cool thank you! I'll take a look.

> I'm happy to close this PR and rebase my fixes on your work in case you'd like to keep working on it.

Well at this point there have already been two attempts to land this. The problem is that it's massive. So maybe we should change strategy and merge a first implementation even if it's incomplete, and work from there. So I'm tempted to try to merge this already. Wdyt?

**Comment by george-hopkins** [comment]
Thank you, it's quite a big addition indeed. However, I think it can serve as a good base due to the good architecture which handles unparsed attributes quite nicely. In addition, I think it will lower the bar for improvements and encourage other developers to contribute the remaining bits over time if the base is already laid out and can serve as an example.

**Comment by little-dude** [comment]
All right. Let merge this in then :)

File: netlink-packet-sock-diag/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[package]
+authors = ["Flier Lu <flier.lu@gmail.com>", "Corentin Henry <corentinhenry@gmail.com>"]
```
Comment (little-dude) [inline netlink-packet-sock-diag/Cargo.toml:2]: @george-hopkins I'd like to release 0.1, and noticed that you haven't added yourself to the authors. Would you like to? If so which name/email would you like to use?

Comment (george-hopkins) [reply netlink-packet-sock-diag/Cargo.toml:2]: Thank you for asking! It's fine by me, I'm glad `sock_diag` is officially supported now :tada:

**Pull request merged**